indexers/src/main/java/org/mousephenotype/cda/indexers/StatisticalResultsIndexer.java

```diff
@@ -471,26 +487,47 @@ private StatisticalResultDTO parseLineResult(ResultSet r) throws SQLException {
 	 * Add the appropriate MP term associations to the document
 	 * This is only used for the embryo data for the moment (2016-04-07)
 	 *
-	 * @param mpTerm the mp term accession id
+	 * @param mpId the mp term accession id
 	 * @param doc the solr document to update
 	 * @throws SQLException if the query fields do not exist
 	 */
-	private void addMpTermData(String mpTerm, StatisticalResultDTO doc) throws SQLException {
+	private void addMpTermData(String mpId, StatisticalResultDTO doc) throws SQLException {
 
 		// Add the appropriate fields for the global MP term
-		if (mpTerm != null) {
+		if (mpId != null) {
+
+			OntologyTermBean mpTerm = mpOntologyService.getTerm(mpId);
+			if (mpTerm != null) {
+				doc.setMpTermId(mpTerm.getId());
+				doc.setMpTermName(mpTerm.getName());
+
+				doc.setTopLevelMpTermId(mpOntologyService.getTopLevelDetail(mpId).getIds());
+				doc.setTopLevelMpTermName(mpOntologyService.getTopLevelDetail(mpId).getNames());
+
+				doc.setIntermediateMpTermId(mpOntologyService.getIntermediatesDetail(mpId).getIds());
+				doc.setIntermediateMpTermName(mpOntologyService.getIntermediatesDetail(mpId).getNames());
+
+				 // mp-anatomy mappings (all MA at the moment)
+                if (mpOntologyService.getAnatomyMappings(mpId) != null){
+                	List<String> anatomyIds = mpOntologyService.getAnatomyMappings(mpId);
+                	for (String id: anatomyIds){
+                		OntologyDAO currentOntologyService = null;
+                		if (id.startsWith("EMAPA")){
+                			currentOntologyService = emapaOntologyService;
+                		} else  if (id.startsWith("MA")){
+                			currentOntologyService = maOntologyService;
+                		}
+                		doc.addAnatomyTermId(id);
```

Comment:
currentOntologyService could be null here if there is a term that doesn't start with MA or EMAPA